File: 0027-remove-element/0027-remove-element.py
class Solution:
  def removeElement(self, nums: List[int], val: int) -> int:
    offset = 0
    for i in range(len(nums)):
      if nums[i] == val:
        offset += 1
      elif offset:
        nums[i - offset] = nums[i]
    return len(nums) - offset

File: 0027-remove-element/0027-remove-element.ts
function removeElement(nums: number[], val: number): number {
  let j = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== val) {
      if (i !== j) nums[j] = nums[i];
      j++;
    }
  }
  return j;  
};